snapcraft/plugins/python.py

```diff
@@ -438,17 +439,13 @@ def list(self, exec_func=None):
         """Return a dict of installed python packages with versions."""
         if not exec_func:
             exec_func = self._exec_func
-        cmd = [*self._runnable, 'list']
+        cmd = [*self._runnable, 'list', '--format=json']
 
         output = exec_func(cmd, env=self._env)
-        package_listing = {}
-        version_regex = re.compile('\((.+)\)')
-        for line in output.splitlines():
-            line = line.split()
-            m = version_regex.search(line[1])
-            package_listing[line[0]] = m.group(1)
-
-        return package_listing
+        return {
+            package['name']: package['version']
+            for package in json.loads(output)
+        }
```

Comment:
I can't really agree with your points to be honest, it's just no tested at the end of the day.

Perhaps we should run python integration tests daily if it's that bad?